crates/block-producer/src/block_builder/prover/mod.rs

```diff
@@ -279,4 +279,20 @@ impl BlockProver {
             new_chain_mmr_root.into(),
         ))
     }
+
+    fn compute_tx_hash(&self, witness: &BlockWitness) -> Digest {
+        let elements: Vec<Felt> = witness
+            .updated_accounts
+            .iter()
+            .flat_map(|(&account_id, update)| {
+                update.transactions.iter().flat_map(move |&tx| {
+                    let mut result = vec![Felt::ZERO, Felt::ZERO, Felt::ZERO, account_id.into()];
```

Comment:
why do we need to add the account id to the elements to hash?

Reply:
Please look into this issue for clarifications: https://github.com/0xPolygonMiden/miden-base/issues/722